homeassistant/components/vevor_heater/coordinator.py

```diff
@@ -0,0 +1,67 @@
+"""Data update coordinators for Vevor BLE Heater integration."""
+
+import logging
+from typing import Optional
+
+from bleak import BLEDevice
+import bluetooth_data_tools
+from vevor_heater_ble.heater import VevorDevice
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_UPDATE_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VevorHeaterUpdateCoordinator(DataUpdateCoordinator[VevorDevice]):
+    """Update coordinator for Vevor BLE Heaters."""
+
+    def __init__(self, hass: HomeAssistant, device: BLEDevice) -> None:
+        """Initialize the coordinator for the given BLE device. Assumes the device corresponds to a Vevor Heater."""
+        self._ble_device = device
+        self._vevor_device = VevorDevice(
+            address=self._ble_device.address,
+            name=f"Vevor {self._ble_device.name or bluetooth_data_tools.short_address(self._ble_device.address)}",
```

Comment:
It would be nicer to make `short_address` a local var.  I try not to use function calls in `f-strings` as it makes it harder to read it takes longer to review.  Keep in mind we have to check every function for blocking I/O in the event loop, and while I know this one, its still a lot of time spent to do that.

Reply:
Makes sense - extracted to a separate local var but kept the or clause as to avoid the extra work when we end up not needing the short name.

While here I also gave address the same treatment.